### ISSUE TITLE
Sanitize file names in database e.g. prevent SQL injections

### ISSUE THREAD

**Issue by inaod568** [ScilifelabDataCentre/dds_web#707]
Check how flask sqlalchemy handles SQL injections.

**Comment by inaod568** [comment]
https://go.snyk.io/rs/677-THP-415/images/Python_Cheatsheet_whitepaper.pdf

**Comment by inaod568** [comment]
@talavis -- It should (as we've discussed previously) just be executing raw SQL in the code that could be an issue, but we need to verify that flask sqlalchemy is not vulnerable to SQL injections.

**Comment by talavis** [comment]
As far as I can tell, it is safe as long as you do not give comparisons as string.
```
*.filter(f"a={val}")
```
is not safe, as raw sql is provided, bypassing the sql injection protections.
```
*.filter(a == val)
```
should be safe.

Thus: as long as you never give raw sql statements in your code, sqlalchemy should be safe.

**Comment by inaod568** [comment]
Ok! 
Does this mean that this is unsafe? https://github.com/ScilifelabDataCentre/dds_web/blob/1083347d3eecd404645ad58c6db2913c2411b6bf/dds_web/api/db_connector.py#L89-L94

**Comment by talavis** [comment]
So I have been investigating the potential for sql injections via the regex. After a lot of different experiments I have the log from mariadb:

When I tried to make a simple injection via my request (`'; SELECT * FROM files; '`:
```
21 Query    SELECT DISTINCT files.subpath AS files_subpath
            FROM files
            WHERE files.project_id = binary(1) 
            AND files.subpath 
            REGEXP '^\'; SELECT * FROM files; \'(/[^/]+)+$'
```
As can be seen in the query, SQLAlchemy has escaped the `'` characters, meaning that it should be safe from SQL injections

Another piece of support: if you run the following in the mariadb CLI
```
SELECT * FROM files; SELECT * FROM projects;
```
It gets interpreted as two queries:
```
5 Query	SELECT * FROM files
5 Query	SELECT * FROM projects
```

I thus believe our regex filter should be safe from SQL injections, meaning that we can close #707.

I also have a pr on the way with the tests I wrote for that endpoint.
